### PR TITLE
bpo-34925: Optimize common case for bisect() argument parsing

### PR DESCRIPTION
**Pull request by rhettinger**
https://bugs.python.org/issue34925

### CODE REVIEW

File: Modules/_bisectmodule.c [hunk 1]
```diff
@@ -8,7 +8,7 @@ Converted to C by Dmitry Vasiliev (dima at hlabs.spb.ru).
 
 _Py_IDENTIFIER(insert);
 
-static Py_ssize_t
+static inline Py_ssize_t
```
Comment (serhiy-storchaka) [inline Modules/_bisectmodule.c:11]: Perhaps you meant `Py_LOCAL_INLINE(Py_ssize_t)`?

File: Modules/_bisectmodule.c [hunk 2]
```diff
@@ -53,9 +53,14 @@ bisect_right(PyObject *self, PyObject *args, PyObject *kw)
     Py_ssize_t index;
     static char *keywords[] = {"a", "x", "lo", "hi", NULL};
 
-    if (!PyArg_ParseTupleAndKeywords(args, kw, "OO|nn:bisect_right",
-        keywords, &list, &item, &lo, &hi))
-        return NULL;
+    if (kw==NULL && PyTuple_GET_SIZE(args)==2) {
```
Comment (serhiy-storchaka) [inline Modules/_bisectmodule.c]: Sorry for be pedantic, but could you please add spaces around `==` for conforming PEP 7? Without spaces the code looks sloppy to me.

File: Modules/_bisectmodule.c [hunk 3]
```diff
@@ -53,9 +53,14 @@ bisect_right(PyObject *self, PyObject *args, PyObject *kw)
...
+        list = PyTuple_GET_ITEM(args, 0);
+        item = PyTuple_GET_ITEM(args, 1);
+    } else {
```
Comment (serhiy-storchaka) [inline Modules/_bisectmodule.c]: For PEP 7 `}` and `else {` should be on separate lines.
```c
    }
    else {
```

**Review by serhiy-storchaka — COMMENTED** [review]
Looks correct, besides few style issues.

I hope this case will be automatically optimized by Argument Clinic in 3.8.

File: Modules/_bisectmodule.c (hunk 2 above)
Comment (rhettinger) [reply Modules/_bisectmodule.c]: IIRC, when multiple terms are present, PEP 7 (and PEP 8) allow for tightening spaces on the inner terms while "adding" spaces around the outermost (lowest-priority) operators.

File: Modules/_bisectmodule.c (hunk 1 above)
Comment (rhettinger) [reply Modules/_bisectmodule.c:11]: Is that being used anymore?   I believe we're using "inline" directly (and some other C99 features) nowadays.

File: Modules/_bisectmodule.c (hunk 2 above)
Comment (serhiy-storchaka) [reply Modules/_bisectmodule.c]: I think this is allowed only for arithmetic operators like `*` in `+`.

> Always put spaces around assignment, Boolean and comparison operators. In expressions using a lot of operators, add spaces around the outermost (lowest-priority) operators.

File: Modules/_bisectmodule.c (hunk 1 above)
Comment (serhiy-storchaka) [reply Modules/_bisectmodule.c:11]: I don't know. Maybe `Py_LOCAL_INLINE` is outdated.

It is expanded to a different form on MSVC.

File: Modules/_bisectmodule.c (hunk 3 above)
Comment (rhettinger) [reply Modules/_bisectmodule.c]: I see that in PEP 7 but it looks weird to me.  I believe we historically allowed "} else {" in multiple places in the code when that reflects how the author thinks about the code.  If you insist, I'll change it, but I like it better as-is.

Comment (serhiy-storchaka) [reply Modules/_bisectmodule.c]: No, I do not insist. I added this comment just for the case if you missed this part of PEP 7 and do not have preferences. I always write this code as two lines in new and modified code, but writing it as a single line LGTM too.

The only thing looks weird to me is messed spaces around `==`.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**